charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml

```diff
@@ -34,14 +34,14 @@ spec:
       {{- end }}
       containers:
         - name: controller
-          image: "{{ .Values.controller.image.registry }}/{{ .Values.controller.image.repository }}:{{ default (include "argo-workflows.defaultTag" .) .Values.controller.image.tag }}"
+          image: "{{- include "argo-workflows.image" (dict "context" . "registry" .Values.controller.image.registry "repository" .Values.controller.image.repository) }}:{{ default (include "argo-workflows.defaultTag" .) .Values.controller.image.tag }}"
```

Comment:
Maybe this can be simplified a little bit by passing `dict "context" . "image" .Values.controller.image`. Then you can use both registry and repository in the template.